### PR TITLE
make batch single threaded

### PR DESCRIPTION
**Pull request by cseed**
polling and k8s watch thread request into main thread
in addition to k8s notifications, periodically poll k8s state

### CODE REVIEW

File: batch/server.py [hunk 1]
```diff
@@ -286,30 +341,27 @@ def kube_event_loop():
...
-            else:
-                log.error(f'kube_event_loop: saw unexpected event_type {event_type} in {event}')
+def polling_event_loop():
+    time.sleep(1)
+    while True:
+        try:
+           r = requests.post('http://127.0.0.1:5000/refresh_k8s_state', timeout=120)
+           r.raise_for_status()
+        except Exception as e:
+            log.error(f'Could not poll due to exception: {e}')
```
Comment (danking) [inline batch/server.py]: does `e` include `r.text`? I've found that the client batch library will `r.raise_for_status()` and the only message I get is that there was a 404 or 500. I'd prefer to include r.text in this message to be sure that we can see whatever information k8s provides in the body of the response.

File: batch/server.py [hunk 2]
```diff
@@ -286,30 +341,27 @@ def kube_event_loop():
     w = kube.watch.Watch()
     stream = w.stream(v1.list_namespaced_pod, 'default')
     for event in stream:
-        event_type = event['type']
         pod = event['object']
```
Comment (danking) [inline batch/server.py]: this is dead code now, right? We make a network request from the Flask thread to get pod state.

File: batch/server.py (hunk 1 above)
Comment (cseed) [reply batch/server.py]: It isn't printed, but it's contained in the exception, which includes the response.  For example:

```
>>> import requests
>>> r = requests.get('http://example.com/foo/bar/baz')
>>> try:
...   r.raise_for_status()
... except Exception as e:
...   x = e
... 
>>> x
HTTPError('404 Client Error: Not Found for url: http://example.com/foo/bar/baz',)
>>> x.response.text
'<!doctype html>\n<html>\n<head>\n    <title>Example Domain</title>\n\n    ...'
```

Fixed.

File: batch/server.py (hunk 2 above)
Comment (cseed) [reply batch/server.py]: No, I send the pod name in the pod_changed request.

Comment (danking) [reply batch/server.py]: ah I see.

**Pull request merged**